lib/ansible/modules/cloud/azure/azure_rm_hdinsightcluster.py

```diff
@@ -0,0 +1,540 @@
+            if self.state == 'absent':
+                self.to_do = Actions.Delete
+            elif self.state == 'present':
+                # get doesn't return storage profile, so can't be compared
+                old_response['properties']['storage_profile'] = self.parameters['properties']['storage_profile']
+                old_response['properties']['cluster_definition'] = self.parameters['properties']['cluster_definition']
+
+                # cluster version returned by get may be longer, for instance "3.6.1000.67"
+                if old_response['properties']['cluster_version'].startswith(self.parameters['properties'].get('cluster_version', '')):
+                    self.parameters['properties']['cluster_version'] = old_response['properties']['cluster_version']
+
+                if (not default_compare(self.parameters, old_response, '', self.results)):
+                    self.results['old_response'] = old_response
+                    self.to_do = Actions.Update
+
+        if (self.to_do == Actions.Create) or (self.to_do == Actions.Update):
+            self.log("Need to Create / Update the Cluster instance")
+
+            if self.check_mode:
+                self.results['changed'] = True
+                return self.results
+
+            response = self.create_update_cluster()
+
+            self.results['changed'] = True
+            self.log("Creation / Update done")
+        elif self.to_do == Actions.Delete:
+            self.log("Cluster instance deleted")
+            self.results['changed'] = True
+
+            if self.check_mode:
+                return self.results
+
+            self.delete_cluster()
+            # make sure instance is actually deleted, for some Azure resources, instance is hanging around
+            # for some time after deletion -- this should be really fixed in Azure.
+            while self.get_cluster():
+                time.sleep(20)
+        else:
+            self.log("Cluster instance unchanged")
+            self.results['changed'] = False
+            response = old_response
+
+        if self.state == 'present':
+            self.results.update(self.format_item(response))
+        return self.results
+
+    def create_update_cluster(self):
+        '''
+        Creates or updates Cluster with the specified configuration.
+
+        :return: deserialized Cluster instance state dictionary
+        '''
+        self.log("Creating / Updating the Cluster instance {0}".format(self.name))
+
+        try:
+            if self.to_do == Actions.Create:
+                response = self.mgmt_client.clusters.create(resource_group_name=self.resource_group,
+                                                            cluster_name=self.name,
+                                                            parameters=self.parameters)
+            else:
+                response = self.mgmt_client.clusters.update(resource_group_name=self.resource_group,
```

Comment:
Does this actually do anything (eg, are there any update operations that actually work on one of these)? By just passing RG/name, I'm guessing this is always a no-op (which is fine, but if it's a create/delete-only resource, might want to warn that updates do nothing...

Reply:
indeed. should be possible to add tags, but seems like I missed it.... :-(
I will add a separate resize cluster functionality as well.